testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -5820,3 +5819,313 @@ fn atlas_stress_integration_test() {
 
     test_observer::clear();
 }
+
+/// Deserializes the `StacksTransaction` objects from `blocks` and returns all those that
+/// match `test_fn`.
+fn select_transactions_where(
+    blocks: &Vec<serde_json::Value>,
+    test_fn: fn(&StacksTransaction) -> bool,
+) -> Vec<StacksTransaction> {
+    let mut result = vec![];
+    for block in blocks {
+        let transactions = block.get("transactions").unwrap().as_array().unwrap();
+        for tx in transactions.iter() {
+            let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+            let tx_bytes = hex_bytes(&raw_tx[2..]).unwrap();
+            let parsed = StacksTransaction::consensus_deserialize(&mut &tx_bytes[..]).unwrap();
+            if test_fn(&parsed) {
+                eprintln!("show-block {:#?}", &block);
+                eprintln!("show-parsed {:#?}", &parsed);
+                result.push(parsed);
+            }
+        }
+    }
+
+    return result;
+}
+
+/// This function will call `next_block_and_wait` until the burnchain height underlying `BitcoinRegtestController`
+/// reaches *at least* `target_height`.
+///
+/// Returns `false` if `next_block_and_wait` times out.
+///
+/// Note: The user cannot assume that the block height reached is *exactly* `target_height`. It
+/// seems that the chain may often advanced `1` block past `target_height`.
+fn run_until_burnchain_height(
+    btc_regtest_controller: &mut BitcoinRegtestController,
+    blocks_processed: &Arc<AtomicU64>,
+    burnchain_height: &Arc<AtomicU64>,
+    target_height: u64,
+) -> bool {
+    let current = burnchain_height.load(Ordering::SeqCst);
+    eprintln!(
+        "run_until_burnchain_height: Issuing block at {}, current burnchain height is ({})",
+        get_epoch_time_secs(),
+        current
+    );
+
+    next_block_and_wait(btc_regtest_controller, &blocks_processed);
+    while burnchain_height.load(Ordering::SeqCst) <= target_height {
+        let current = burnchain_height.load(Ordering::SeqCst);
+        eprintln!(
+            "run_until_burnchain_height: Issuing block at {}, current burnchain height is ({})",
+            get_epoch_time_secs(),
+            current
+        );
+        let next_result = next_block_and_wait(btc_regtest_controller, &blocks_processed);
+        if !next_result {
+            return false;
+        }
+    }
+
+    true
+}
+
+/// This test checks that the block limit is changed at Stacks 2.05. We lower the allowance, and
+/// check that we can 1) afford the function call before the target height and
+/// 2) cannot afford the function call after the target height.
+#[test]
+#[ignore]
+fn test_cost_limit_switch_version205() {
```

Comment:
Can you move this test to the `epoch_205` module?

And also add a line to `.github/workflows/bitcoin-tests.yml` to add this test to the continuous integrations.